components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -1064,3 +1064,118 @@ func (backend *ESClient) SetDailyLatestToFalseForControlIndex(ctx context.Contex
 		Do(ctx)
 	return errors.Wrap(err, "SetDailyLatestToFalseForControlIndex")
 }
+
+func (backend *ESClient) GetNodesDayLatestTrue(ctx context.Context) ([]relaxting.NodesUpgradation, error) {
+	var nodes []relaxting.NodesUpgradation
+	time90daysAgo := time.Now().Add(-24 * time.Hour * 90)
+	indices, err := GetLast90DaysIndices(time90daysAgo, time.Now())
+	if err != nil {
+		return nil, err
+	}
+
+	boolQuery := elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery("day_latest", true)).
+		Must(elastic.NewTermQuery("daily_latest", true))
+
+	fsc := elastic.NewFetchSourceContext(true).Include(
+		"node_uuid",
+		"end_time",
+		"day_latest",
+	)
+
+	searchSource := elastic.NewSearchSource().
+		FetchSourceContext(fsc).
+		Query(boolQuery).
+		Size(10000).Sort("end_time", false)
+
+	searchResult, err := backend.client.Search().
+		SearchSource(searchSource).
+		Index(indices).
+		FilterPath(
+			"took",
+			"hits.total",
+			"hits.hits._id",
+			"hits.hits._source",
+			"hits.hits.inner_hits").
+		Do(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if searchResult.TotalHits() > 0 {
+		logrus.Printf("Found a total of %d ESInSpecReport\n", searchResult.TotalHits())
+		// Iterate through results
+		for _, hit := range searchResult.Hits.Hits {
+			var node relaxting.NodesUpgradation
+			if hit.Source != nil {
+				err := json.Unmarshal(hit.Source, &node)
+				if err != nil {
+					logrus.Errorf("Received error while unmarshling %+v", err)
+				}
+
+			}
+			nodes = append(nodes, node)
```

Comment:
Are we getting unique nodes here or all the nodes?

Reply:
@kalroy all the nodes with `day_letest` and `daily_latest` true

Reply:
That's fine but unique nodes? a node (node-1) can have these flags set on 10th July as well as on 4th July, so will nodes array will be [node-1] or [node-1, node-1] ? @dave-augustus